### PR TITLE
KAFKA-4211: Update system tests to use the new consumer by default

### PR DESCRIPTION
**Pull request by vahidhashemian**
Update system test method signatures and method calls to use the new consumer by default.

### CODE REVIEW

**Comment by vahidhashemian** [comment]
This patch passed system tests ([here](https://jenkins.confluent.io/job/system-test-kafka-branch-builder/598/)).

**Comment by vahidhashemian** [comment]
@granders @hachikuji I'd appreciate your feedback on this PR.

File: tests/kafkatest/sanity_checks/test_performance_services.py [hunk 1]
```diff
@@ -37,12 +37,12 @@ def setUp(self):
 
     # We are keeping 0.8.2 here so that we don't inadvertently break support for it. Since this is just a sanity check,
     # the overhead should be manageable.
-    @parametrize(version=str(LATEST_0_8_2))
+    @parametrize(version=str(LATEST_0_8_2), new_consumer=False)
     @parametrize(version=str(LATEST_0_9), new_consumer=False)
     @parametrize(version=str(LATEST_0_9), new_consumer=True)
```
Comment (hachikuji) [inline tests/kafkatest/sanity_checks/test_performance_services.py]: minor: seems like we got rid of the `new_consumer` parameter in the other cases below when using the default.

Comment (vahidhashemian) [reply tests/kafkatest/sanity_checks/test_performance_services.py]: Thanks. I tried to follow suit here, as the `False` default was also explicitly provided before. I'll update it and remove the parameter when it is the default. I hope I did not misunderstand your point.

Comment (hachikuji) [reply tests/kafkatest/sanity_checks/test_performance_services.py]: Oh, I see. Well, either way works for me.

Comment (vahidhashemian) [reply tests/kafkatest/sanity_checks/test_performance_services.py]: Thanks. I removed the redundant parameter. I noticed in other places in the code this redundancy does not exist.

File: tests/kafkatest/tests/core/mirror_maker_test.py [hunk 2]
```diff
@@ -45,10 +45,10 @@ def __init__(self, test_context):
         self.producer = VerifiableProducer(test_context, num_nodes=1, kafka=self.source_kafka, topic=self.topic,
                                            throughput=1000)
         self.mirror_maker = MirrorMaker(test_context, num_nodes=1, source=self.source_kafka, target=self.target_kafka,
-                                        whitelist=self.topic, offset_commit_interval_ms=1000)
+                                        whitelist=self.topic, new_consumer=False, offset_commit_interval_ms=1000)
```
Comment (hachikuji) [inline tests/kafkatest/tests/core/mirror_maker_test.py]: Seems like we override this option in the tests anyway, so maybe we could just use the default? Same for the `ConsoleConsumer` below.

**Comment by hachikuji** [comment]
One minor question. Otherwise, LGTM.

File: tests/kafkatest/tests/core/mirror_maker_test.py (hunk 2 above)
Comment (vahidhashemian) [reply tests/kafkatest/tests/core/mirror_maker_test.py]: Yes, you are correct. I'll remove the unnecessary option from both calls. Thanks.

File: tests/kafkatest/tests/connect/connect_distributed_test.py [hunk 3]
```diff
@@ -418,7 +418,7 @@ def test_bounce(self, clean):
         if not success:
             self.mark_for_collect(self.cc)
             # Also collect the data in the topic to aid in debugging
-            consumer_validator = ConsoleConsumer(self.test_context, 1, self.kafka, self.source.topic, consumer_timeout_ms=1000, print_key=True)
+            consumer_validator = ConsoleConsumer(self.test_context, 1, self.kafka, self.source.topic, new_consumer=False, consumer_timeout_ms=1000, print_key=True)
```
Comment (hachikuji) [inline tests/kafkatest/tests/connect/connect_distributed_test.py]: Sorry, one more comment. Can you think of any reason why this shouldn't use the new consumer?

Comment (vahidhashemian) [reply tests/kafkatest/tests/connect/connect_distributed_test.py]: Hmm, I'm not sure. To be honest, I did not pay close attention to the logic behind the tests. I just made sure if it used the old consumer before, it should still use the old consumer. It's not obvious to me from the test itself.

Comment (hachikuji) [reply tests/kafkatest/tests/connect/connect_distributed_test.py]: Not required if you don't want to, but I think it's reasonable/safe to change here. I think the only time we should use the old consumer is when it is the target under test. Here we appear to just need _some_ consumer for additional validation.

Comment (vahidhashemian) [reply tests/kafkatest/tests/connect/connect_distributed_test.py]: Sure, it makes sense. I'll remove it in the next update. Thanks for clarifying.

**Comment by hachikuji** [comment]
LGTM. I kicked off a build of the Connect distributed test to be sure we didn't break anything. I'll merge after it finishes.

**Comment by vahidhashemian** [comment]
Thanks.

**Pull request closed without merging**